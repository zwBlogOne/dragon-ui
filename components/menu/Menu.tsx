import React, { Component, Children, cloneElement, ReactElement } from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import PropsType, { childPropsType } from './PropsType';
import MenuContext, { menuKeys, keysType } from './menu-context';

class Menu extends Component<PropsType, any> {
  static defaultProps = {
    prefixCls: 'ui-menu',
    mode: 'inline',
    theme: 'light',
    inlineIndent: 24,
    inlineCollapsed: false,
    defaultOpenKeys: [],
    defaultSelectedKeys: [],
  };

  static contextTypes = {
    siderCollapsed: PropTypes.bool,
  };

  static SubMenu;
  static Item;
  static Divider;

  static getDerivedStateFromProps(props) {
    const state = {} as keysType;
    if ('openKeys' in props) {
      state.openKeys = props.openKeys;
    }
    if ('selectedKeys' in props) {
      state.selectedKeys = props.selectedKeys;
    }

    return Object.keys(state).length > 0 ? state : null;
  }

  menuKeys: any;

  constructor(props) {
    super(props);
    const { defaultOpenKeys, defaultSelectedKeys } = props;
    this.state = {
      openKeys: defaultOpenKeys,
      selectedKeys: defaultSelectedKeys,
    };
    // 每个实例都有自己的menuKeys，如果都指向一个menuKeys引用会有问题
    this.menuKeys = { ...menuKeys };
    this.menuKeys.toggleSelectedKeys = this.toggleSelectedKeys;
    this.menuKeys.toggleOpenKeys = this.toggleOpenKeys;
  }

  toggleSelectedKeys = (itemKey) => {
    if (!('selectedKeys' in this.props)) {
      this.setState({
        selectedKeys: [itemKey],
      });
    }
  }

  toggleOpenKeys = (subMenuKey) => {
    const { openKeys } = this.state;
    const { onOpenChange } = this.props;
    const newOpenKeys = [...openKeys];

    const keyIndex = openKeys.indexOf(subMenuKey);
    if (keyIndex > -1) {
      newOpenKeys.splice(keyIndex, 1);
    } else {
      newOpenKeys.push(subMenuKey);
    }

    if (onOpenChange) {
      onOpenChange(newOpenKeys);
    }
    if (!('openKeys' in this.props)) {
      this.setState({
        openKeys: newOpenKeys,
      });
    }
  }

  renderChildren() {
    const { children, inlineIndent, inlineCollapsed, mode } = this.props;
    const { siderCollapsed } = this.context;

    const childProps: childPropsType = {
      mode,
      inlineIndent,
      inlineCollapsed: inlineCollapsed || siderCollapsed,
    };
    return Children.map(children, (child, index) => {
      const c: ReactElement<any> = child as ReactElement<any>;
      const key = (child as ReactElement<any>).key;

      childProps.subMenuKey = key || `0-${index}`;
      childProps.itemKey = key || `0-${index}`;
      return cloneElement(c, childProps);
    });
  }

  render() {
    const {
      size, theme, mode, className, style, prefixCls, inlineCollapsed,
    } = this.props;
    const { siderCollapsed } = this.context;

    const { openKeys, selectedKeys } = this.state;
    const cls = classnames({
      [prefixCls!]: true,
      [`${prefixCls}-${theme}`]: true,
      [`${prefixCls}-${mode}`]: true,
      [`${prefixCls}-collapsed`]: !!siderCollapsed || !!inlineCollapsed,
      [`size-${size}`]: !!size,
      [className!]: !!className,
    });

    const newMenuKeys: keysType = {
      ...this.menuKeys,
      openKeys,
      selectedKeys,
    };

    return (
      <ul role="menu" className={cls} style={style}>
        <MenuContext.Provider value={newMenuKeys}>
          {this.renderChildren()}
        </MenuContext.Provider>
      </ul>
    );
  }
}

export default Menu;
