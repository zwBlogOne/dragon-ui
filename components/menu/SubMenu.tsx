import React, { Component, Children, cloneElement, ReactElement } from 'react';
import classnames from 'classnames';
import dom from '../utils/dom';
import events from '../utils/events';
import { SubMenuProps, styleType, childPropsType } from './PropsType';
import MenuContext from './menu-context';

export class SubMenu extends Component<SubMenuProps, any> {
  static defaultProps = {
    prefixCls: 'ui-menu',
    title: '',
    level: 1,
    style: {},
    openKeys: [],
  };

  subTitle: any;
  sub: any;

  constructor(props) {
    super(props);
    this.state = {
      collapsedSubVisible: false,
      collapsedSubAnimation: '',
    };
  }

  toggleSubMenuOpen = (e) => {
    e.stopPropagation();
    const { subMenuKey } = this.props;

    this.props.toggleOpenKeys(subMenuKey);
  }

  renderChildren() {
    const {
      children, level, inlineIndent, mode, prefixCls, subMenuKey,
    } = this.props;
    const childProps: childPropsType = {
      mode,
      level: level + 1,
      inlineIndent,
      prefixCls,
    };
    return Children.map(children, (child, index) => {
      const c: ReactElement<any> = child as ReactElement<any>;
      const key = (child as ReactElement<any>).key;

      childProps.itemKey = key || `${subMenuKey}-${level}-${index}`;
      childProps.subMenuKey = key || `${subMenuKey}-${level}-${index}`;

      return cloneElement(c, childProps);
    });
  }

  getSubHeight() {
    if (!this.sub) {
      return;
    }
    const childs = [...this.sub.children];
    let marginBottom = 0;

    if (childs[0]) {
      marginBottom = parseFloat(dom.getStyleComputedProperty(childs[0], 'marginBottom'));
    }

    const height = childs.reduce((res, next) => {
      res += (next.offsetHeight + marginBottom);
      return res;
    }, marginBottom / 2);

    return height;
  }

  setSubHeight(prevProps) {
    if (!this.sub) {
      return;
    }
    const { openKeys: lastOpenKeys } = prevProps;
    const { openKeys, subMenuKey } = this.props;

    const keyIndex = openKeys.indexOf(subMenuKey);
    const keysLength = openKeys.length;
    if (keyIndex > -1) {
      if (keysLength > 1 && keyIndex < keysLength - 1 || keysLength < lastOpenKeys.length) {
        // 如果不是最后一级子菜单，或者嵌套的子菜单被收起，当前子菜单高度自适应
        this.sub.style.height = 'auto';
      } else {
        // 否则，设置具体的高度产生过渡动画
        const height = this.getSubHeight();
        this.sub.style.height = height + 'px';
      }
    } else {
      const height = this.getSubHeight();
      this.sub.style.height = height + 'px';

      setTimeout(() => {
        this.sub.style.height = 0;
      }, 0);
    }
  }

  slideUp() {
    this.setState({
      collapsedSubVisible: true,
      collapsedSubAnimation: 'up',
    });
  }

  slideDown() {
    this.setState({
      collapsedSubVisible: true,
      collapsedSubAnimation: 'down',
    });
  }

  onSubAnimationEnd = () => {
    const { subMenuKey, openKeys } = this.props;
    const isOpen = openKeys.indexOf(subMenuKey) > -1;
    this.setState({
      collapsedSubVisible: isOpen,
    });
  }

  onClickOutSide = (e) => {
    const { target } = e;
    const { subMenuKey, openKeys } = this.props;
    if (this.subTitle.contains(target)) {
      return;
    }
    if (!this.sub.contains(target) && openKeys.indexOf(subMenuKey) > -1) {
      this.props.toggleOpenKeys(subMenuKey);
    }
  }

  componentDidMount() {
    const { openKeys, inlineCollapsed } = this.props;
    if (openKeys.length > 0) {
      if (!inlineCollapsed) {
        this.setSubHeight({ openKeys: [] });
      }
    }
    if (inlineCollapsed) {
      events.on(document, 'click', this.onClickOutSide);
    }
  }

  componentWillReceiveProps(nextProps) {
    const { inlineCollapsed } = nextProps;

    if (!inlineCollapsed) {
      if (this.props.inlineCollapsed) {
        events.off(document, 'click', this.onClickOutSide);
      }
      return;
    }
    if (!this.props.inlineCollapsed) {
      events.on(document, 'click', this.onClickOutSide);
    }
    const { subMenuKey, openKeys } = this.props;
    const isOpenNow = openKeys.indexOf(subMenuKey) > -1;
    const isOpenNext = nextProps.openKeys.indexOf(subMenuKey) > -1;

    if (!isOpenNow && isOpenNext || (!this.props.inlineCollapsed && isOpenNow)) {
      // 展开菜单
      this.slideDown();
    } else if (isOpenNow && !isOpenNext) {
      // 收起菜单
      this.slideUp();
    }
  }

  componentDidUpdate(prevProps) {
    const { inlineCollapsed } = this.props;
    if (!inlineCollapsed) {
      this.setSubHeight(prevProps);
    }
  }

  componentWillUnmount() {
    const { inlineCollapsed } = this.props;
    if (inlineCollapsed) {
      events.off(document, 'click', this.onClickOutSide);
    }
  }

  render() {
    const {
      title, level, mode, style, inlineIndent,
      prefixCls, openKeys, subMenuKey, inlineCollapsed,
    } = this.props;
    const { collapsedSubVisible, collapsedSubAnimation } = this.state;

    const subMenuStyle: styleType = {};
    if (mode === 'inline') {
      subMenuStyle.paddingLeft = level * inlineIndent;
    }
    const isOpen = openKeys.indexOf(subMenuKey) > -1;
    const cls = classnames(`${prefixCls}-submenu`, {
      open: isOpen,
    });
    let subStyle: React.CSSProperties = {
      display: 'block',
    };
    let subCls = `${prefixCls}-submenu-sub`;
    if (inlineCollapsed) {
      subStyle = {
        display: collapsedSubVisible ? 'block' : 'none',
      };
      subCls = classnames(`${prefixCls}-submenu-sub`, {
        [`slide-${collapsedSubAnimation}`]: !!collapsedSubAnimation,
      });
    }

    return (
      <li className={cls} style={style}>
        <div
          ref={(subTitle) => { this.subTitle = subTitle; }}
          onClick={this.toggleSubMenuOpen}
          style={subMenuStyle}
          className={`${prefixCls}-submenu-title`}
        >
          {title}
          <i className={`${prefixCls}-submenu-arrow`} />
        </div>
        <ul
          ref={(sub) => { this.sub = sub; }}
          className={subCls}
          style={subStyle}
          onAnimationEnd={this.onSubAnimationEnd}
        >
          {this.renderChildren()}
        </ul>
      </li>
    );
  }
}

export default function SubMenuConsumer(props) {
  return (
    <MenuContext.Consumer>
      {
        (menuKeys) => (
          <SubMenu
            {...props}
            openKeys={menuKeys.openKeys}
            toggleOpenKeys={menuKeys.toggleOpenKeys}
          />
        )}
    </MenuContext.Consumer>
  );
}
