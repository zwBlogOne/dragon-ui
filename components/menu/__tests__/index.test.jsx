import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Menu from '../index';
import { SubMenu } from '../SubMenu';

describe('Menu', () => {
  it('renders horizontal Menu correctly', () => {
    const wrapper = render(
      <Menu mode="horizontal" defaultSelectedKeys={['a']}>
        <Menu.Item key="a">意健险</Menu.Item>
        <Menu.Item key="b">健康险个险</Menu.Item>
        <Menu.Item key="c">雇主责任险</Menu.Item>
        <Menu.Item key="d">运营后台管理</Menu.Item>
        <Menu.Item key="e">公共功能</Menu.Item>
        <Menu.Item key="f">询报价</Menu.Item>
      </Menu>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders inline Menu with SubMenu correctly', () => {
    const wrapper = render(
      <Menu>
        <Menu.SubMenu title="新契约">
          <Menu.Item>投保单复核</Menu.Item>
          <Menu.Item>在线投保单管理</Menu.Item>
          <Menu.Item>投保单录入</Menu.Item>
          <Menu.Item>新增计划</Menu.Item>
        </Menu.SubMenu>
        <Menu.SubMenu title="核保">
          <Menu.Item>核保权限定义</Menu.Item>
          <Menu.Item>核保权限分配</Menu.Item>
          <Menu.Item>规则配置</Menu.Item>
          <Menu.Item>人工核保</Menu.Item>
        </Menu.SubMenu>
        <Menu.SubMenu title="批改">
          <Menu.Item>批改新增</Menu.Item>
          <Menu.Item>批改复核</Menu.Item>
          <Menu.Item>批改回退</Menu.Item>
          <Menu.Item>批改共享池</Menu.Item>
        </Menu.SubMenu>
      </Menu>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders nested SubMenus correctly', () => {
    const wrapper = render(
      <Menu>
        <Menu.SubMenu title="理赔">
          <Menu.Item>报案</Menu.Item>
          <Menu.Item>任务分配</Menu.Item>
          <Menu.SubMenu title="理赔工作流">
            <Menu.Item>休假维护</Menu.Item>
            <Menu.Item>时效维护</Menu.Item>
          </Menu.SubMenu>
        </Menu.SubMenu>
        <Menu.SubMenu title="新契约">
          <Menu.Item>投保单复核</Menu.Item>
          <Menu.Item>在线投保单管理</Menu.Item>
          <Menu.Item>投保单录入</Menu.Item>
          <Menu.Item>新增计划</Menu.Item>
        </Menu.SubMenu>
        <Menu.SubMenu title="核保">
          <Menu.Item>核保权限定义</Menu.Item>
          <Menu.Item>核保权限分配</Menu.Item>
          <Menu.Item>规则配置</Menu.Item>
          <Menu.Item>人工核保</Menu.Item>
        </Menu.SubMenu>
      </Menu>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders dark Menu correctly', () => {
    const wrapper = render(
      <Menu theme="dark">
        <Menu.SubMenu title="新契约">
          <Menu.Item>投保单复核</Menu.Item>
          <Menu.Item>在线投保单管理</Menu.Item>
          <Menu.Item>投保单录入</Menu.Item>
          <Menu.Item>新增计划</Menu.Item>
        </Menu.SubMenu>
        <Menu.SubMenu title="核保">
          <Menu.Item>核保权限定义</Menu.Item>
          <Menu.Item>核保权限分配</Menu.Item>
          <Menu.Item>规则配置</Menu.Item>
          <Menu.Item>人工核保</Menu.Item>
        </Menu.SubMenu>
        <Menu.SubMenu title="批改">
          <Menu.Item>批改新增</Menu.Item>
          <Menu.Item>批改复核</Menu.Item>
          <Menu.Item>批改回退</Menu.Item>
          <Menu.Item>批改共享池</Menu.Item>
        </Menu.SubMenu>
      </Menu>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Menu with Divider correctly', () => {
    const wrapper = render(
      <Menu theme="dark">
        <Menu.Item>投保单复核</Menu.Item>
        <Menu.Item>在线投保单管理</Menu.Item>
        <Menu.Divider />
        <Menu.Item>投保单录入</Menu.Item>
        <Menu.Item>新增计划</Menu.Item>
      </Menu>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders collpased Menu correctly', () => {
    const wrapper = render(
      <Menu inlineCollapsed>
        <Menu.Item>投保单复核</Menu.Item>
        <Menu.Item>在线投保单管理</Menu.Item>
        <Menu.Divider />
        <Menu.Item>投保单录入</Menu.Item>
        <Menu.Item>新增计划</Menu.Item>
      </Menu>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when toggling collapsed status', () => {
    const wrapper = shallow(
      <Menu>
        <Menu.Item>投保单复核</Menu.Item>
        <Menu.Item>在线投保单管理</Menu.Item>
        <Menu.Item>投保单录入</Menu.Item>
        <Menu.Item>新增计划</Menu.Item>
      </Menu>
    );

    wrapper.setProps({ inlineCollapsed: true });

    expect(wrapper.find('.ui-menu').hasClass('ui-menu-collapsed')).toBeTruthy();
  });

  it('behaves correctly when toggling subMenu', () => {
    const subMenuKey = 'key';
    const wrapper = shallow(
      <SubMenu openKeys={[]} subMenuKey={subMenuKey}>
        <Menu.Item>投保单复核</Menu.Item>
        <Menu.Item>在线投保单管理</Menu.Item>
        <Menu.Item>投保单录入</Menu.Item>
        <Menu.Item>新增计划</Menu.Item>
      </SubMenu>
    );

    wrapper.setProps({ openKeys: [subMenuKey] });

    expect(wrapper.find('li').hasClass('open')).toBeTruthy();
  });
});
