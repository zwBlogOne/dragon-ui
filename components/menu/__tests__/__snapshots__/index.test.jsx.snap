// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders Menu with Divider correctly 1`] = `
<ul
  class="ui-menu ui-menu-dark ui-menu-inline"
  role="menu"
>
  <li
    class="ui-menu-item"
    role="menuitem"
    style="padding-left:24px"
  >
    投保单复核
  </li>
  <li
    class="ui-menu-item"
    role="menuitem"
    style="padding-left:24px"
  >
    在线投保单管理
  </li>
  <li
    class="ui-menu-item-divider"
  />
  <li
    class="ui-menu-item"
    role="menuitem"
    style="padding-left:24px"
  >
    投保单录入
  </li>
  <li
    class="ui-menu-item"
    role="menuitem"
    style="padding-left:24px"
  >
    新增计划
  </li>
</ul>
`;

exports[`Menu renders collpased Menu correctly 1`] = `
<ul
  class="ui-menu ui-menu-light ui-menu-inline ui-menu-collapsed"
  role="menu"
>
  <li
    class="ui-menu-item"
    role="menuitem"
    style="padding-left:24px"
  >
    投保单复核
  </li>
  <li
    class="ui-menu-item"
    role="menuitem"
    style="padding-left:24px"
  >
    在线投保单管理
  </li>
  <li
    class="ui-menu-item-divider"
  />
  <li
    class="ui-menu-item"
    role="menuitem"
    style="padding-left:24px"
  >
    投保单录入
  </li>
  <li
    class="ui-menu-item"
    role="menuitem"
    style="padding-left:24px"
  >
    新增计划
  </li>
</ul>
`;

exports[`Menu renders dark Menu correctly 1`] = `
<ul
  class="ui-menu ui-menu-dark ui-menu-inline"
  role="menu"
>
  <li
    class="ui-menu-submenu"
  >
    <div
      class="ui-menu-submenu-title"
      style="padding-left:24px"
    >
      新契约
      <i
        class="ui-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ui-menu-submenu-sub"
      style="display:block"
    >
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        投保单复核
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        在线投保单管理
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        投保单录入
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        新增计划
      </li>
    </ul>
  </li>
  <li
    class="ui-menu-submenu"
  >
    <div
      class="ui-menu-submenu-title"
      style="padding-left:24px"
    >
      核保
      <i
        class="ui-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ui-menu-submenu-sub"
      style="display:block"
    >
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        核保权限定义
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        核保权限分配
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        规则配置
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        人工核保
      </li>
    </ul>
  </li>
  <li
    class="ui-menu-submenu"
  >
    <div
      class="ui-menu-submenu-title"
      style="padding-left:24px"
    >
      批改
      <i
        class="ui-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ui-menu-submenu-sub"
      style="display:block"
    >
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        批改新增
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        批改复核
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        批改回退
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        批改共享池
      </li>
    </ul>
  </li>
</ul>
`;

exports[`Menu renders horizontal Menu correctly 1`] = `
<ul
  class="ui-menu ui-menu-light ui-menu-horizontal"
  role="menu"
>
  <li
    class="ui-menu-item active"
    role="menuitem"
  >
    意健险
  </li>
  <li
    class="ui-menu-item"
    role="menuitem"
  >
    健康险个险
  </li>
  <li
    class="ui-menu-item"
    role="menuitem"
  >
    雇主责任险
  </li>
  <li
    class="ui-menu-item"
    role="menuitem"
  >
    运营后台管理
  </li>
  <li
    class="ui-menu-item"
    role="menuitem"
  >
    公共功能
  </li>
  <li
    class="ui-menu-item"
    role="menuitem"
  >
    询报价
  </li>
</ul>
`;

exports[`Menu renders inline Menu with SubMenu correctly 1`] = `
<ul
  class="ui-menu ui-menu-light ui-menu-inline"
  role="menu"
>
  <li
    class="ui-menu-submenu"
  >
    <div
      class="ui-menu-submenu-title"
      style="padding-left:24px"
    >
      新契约
      <i
        class="ui-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ui-menu-submenu-sub"
      style="display:block"
    >
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        投保单复核
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        在线投保单管理
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        投保单录入
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        新增计划
      </li>
    </ul>
  </li>
  <li
    class="ui-menu-submenu"
  >
    <div
      class="ui-menu-submenu-title"
      style="padding-left:24px"
    >
      核保
      <i
        class="ui-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ui-menu-submenu-sub"
      style="display:block"
    >
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        核保权限定义
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        核保权限分配
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        规则配置
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        人工核保
      </li>
    </ul>
  </li>
  <li
    class="ui-menu-submenu"
  >
    <div
      class="ui-menu-submenu-title"
      style="padding-left:24px"
    >
      批改
      <i
        class="ui-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ui-menu-submenu-sub"
      style="display:block"
    >
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        批改新增
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        批改复核
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        批改回退
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        批改共享池
      </li>
    </ul>
  </li>
</ul>
`;

exports[`Menu renders nested SubMenus correctly 1`] = `
<ul
  class="ui-menu ui-menu-light ui-menu-inline"
  role="menu"
>
  <li
    class="ui-menu-submenu"
  >
    <div
      class="ui-menu-submenu-title"
      style="padding-left:24px"
    >
      理赔
      <i
        class="ui-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ui-menu-submenu-sub"
      style="display:block"
    >
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        报案
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        任务分配
      </li>
      <li
        class="ui-menu-submenu"
      >
        <div
          class="ui-menu-submenu-title"
          style="padding-left:48px"
        >
          理赔工作流
          <i
            class="ui-menu-submenu-arrow"
          />
        </div>
        <ul
          class="ui-menu-submenu-sub"
          style="display:block"
        >
          <li
            class="ui-menu-item"
            role="menuitem"
            style="padding-left:72px"
          >
            休假维护
          </li>
          <li
            class="ui-menu-item"
            role="menuitem"
            style="padding-left:72px"
          >
            时效维护
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    class="ui-menu-submenu"
  >
    <div
      class="ui-menu-submenu-title"
      style="padding-left:24px"
    >
      新契约
      <i
        class="ui-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ui-menu-submenu-sub"
      style="display:block"
    >
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        投保单复核
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        在线投保单管理
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        投保单录入
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        新增计划
      </li>
    </ul>
  </li>
  <li
    class="ui-menu-submenu"
  >
    <div
      class="ui-menu-submenu-title"
      style="padding-left:24px"
    >
      核保
      <i
        class="ui-menu-submenu-arrow"
      />
    </div>
    <ul
      class="ui-menu-submenu-sub"
      style="display:block"
    >
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        核保权限定义
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        核保权限分配
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        规则配置
      </li>
      <li
        class="ui-menu-item"
        role="menuitem"
        style="padding-left:48px"
      >
        人工核保
      </li>
    </ul>
  </li>
</ul>
`;
